spec/models/project_spec.rb

```diff
@@ -248,4 +248,53 @@
       it_behaves_like 'respecting group assignment settings'
     end
   end
+
+  describe 'countable required_project_storage' do
+    describe '#required_storage' do
+      it 'counts collections' do
+        expect(project).to receive(:count_for).with(:required_project_storage).and_call_original
+        storage_hash = project.required_storage
+        expect(storage_hash).to eq({"attributes.attachments"=>0, :total=>0})
+      end
+
+      it 'counts only once before caching' do
+        expect(project).to receive(:count_for).with(:required_project_storage).once
+        project.required_storage
+        project.required_storage
+      end
+    end
+
+    describe '#total_projects_size' do
+      let(:projects) { FactoryGirl.build_list(:project, 3) }
+      before do
+        projects.each(&:save!)
+        allow(Project).to receive(:all).and_return(projects)
+
+        allow(projects[0]).to receive(:count_for).and_return(
+            {"attributes.attachments"=> 23543, :total=> 23543}
+        )
+        allow(projects[1]).to receive(:count_for).and_return(
+            {"attributes.attachments"=> 2, label_repository: 2412345, :total=> 2412347}
```

Comment:
Redundant curly braces around a hash parameter.<br>Indent the first parameter one step more than the previous line.<br>Surrounding space missing for operator '=>'.<br>Space inside { missing.<br>Space inside } missing.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.